phosphor-auditlog/auditLogMgr.cpp

```diff
@@ -0,0 +1,123 @@
+#include "auditLogMgr.hpp"
+
+#include "auditLogParser.hpp"
+#include "auditLogUtils.hpp"
+
+#include <fcntl.h>
+#include <libaudit.h>
+
+#include <phosphor-logging/lg2.hpp>
+#include <sdeventplus/event.hpp>
+#include <sdeventplus/source/event.hpp>
+#include <xyz/openbmc_project/Common/File/error.hpp>
+
+#include <cstring>
+#include <string>
+
+namespace phosphor
+{
+namespace auditlog
+{
+
+sdbusplus::message::unix_fd ALManager::parseAuditLog()
+{
+#ifdef AUDITLOG_KEEP_JSONFILE
+    ALParseFile parsedFile("/tmp/auditLog.json");
+#else
+    ALParseFile parsedFile;
+#endif
+
+    // Initialize to parse full audit log
+    ALParseAll auditParser(parsedFile);
+
+    lg2::debug("Method ParseAuditLog filePath={FILEPATH}", "FILEPATH",
+               parsedFile.getPath());
+
+    // Parse all the events
+    auditParser.doParse();
+
+    // Get file descriptor to return
+    auto fd = openParseFD(parsedFile);
+
+    if (fd >= 0)
```

Comment:
what if the fd is less than zero?

Reply:
openParseFD() throws on error, as you noted later in the review.

Would a comment here be helpful for clarity? Or should I also handle the error here for clarity?

Reply:
Yea, best to either add a comment, handle the error here, or remove the >= 0 check.